crypto/s2n_ecc_evp.c

```diff
@@ -335,19 +348,16 @@ int s2n_ecc_evp_write_params_point(struct s2n_ecc_evp_params *ecc_evp_params, st
 
 #if MODERN_EC_SUPPORTED
     struct s2n_blob point_blob = {0};
-    uint8_t *encoded_point = NULL;
+    DEFER_CLEANUP(uint8_t *encoded_point = NULL, OPENSSL_free_pointer);
 
     size_t size = EVP_PKEY_get1_tls_encodedpoint(ecc_evp_params->evp_pkey, &encoded_point);
     if (size != ecc_evp_params->negotiated_curve->share_size) {
-        OPENSSL_free(encoded_point);
         S2N_ERROR(S2N_ERR_ECDHE_SERIALIZING);
-    } 
-    else {
-        point_blob.data = s2n_stuffer_raw_write(out, ecc_evp_params->negotiated_curve->share_size);
-        notnull_check(point_blob.data);
-        memcpy_check(point_blob.data, encoded_point, size);
-        OPENSSL_free(encoded_point);
     }
+
+    point_blob.data = s2n_stuffer_raw_write(out, ecc_evp_params->negotiated_curve->share_size);
+    notnull_check(point_blob.data);
+    memcpy_check(point_blob.data, encoded_point, size);
```

Comment:
After the copy from `encoded_point` to `point_blob.data`, we must release the memory held by `encoded_point`. With the addition of the DEFER_POINTER_CLEANUP_FUNC for OPENSSL_free, can we verify that the memory is indeed cleaned-up immediately?